app/routes/profile.tsx

```diff
@@ -0,0 +1,70 @@
+import { json, redirect } from "@remix-run/node";
+import type { LoaderFunction } from "@remix-run/node";
+import type { FC } from "react";
+import {
+  commitSession,
+  destroySession,
+  getCurrentUser,
+  getSession,
+} from "~/session.server";
+import { useLoaderData } from "@remix-run/react";
+import RootLayout from "~/components/RootLayout";
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const session = await getSession(request.headers.get("Cookie"));
+  const currentUser = await getCurrentUser(session);
+
+  if (!currentUser) {
+    // Redirect to the login page if needed
+    session.unset("idToken");
+    return redirect("/login", {
+      headers: {
+        "Set-Cookie": await destroySession(session),
+      },
+    });
+  }
+
+  const data: User = {
+    displayName: currentUser.name,
+    avatar: currentUser.picture,
+    email: currentUser.email,
+  };
+
+  return json(data, {
+    headers: { "Set-Cookie": await commitSession(session) },
+  });
+};
+
+type User = {
+  displayName: string;
+  avatar?: string;
+  email?: string;
+};
```

Comment:
Nitpick (Non-blocking): Would it make sense to move `User` above the loader since the loader references the `User` type?

Reply:
Good call, I'll actually move it to the `types.ts` file 👍